analyzers/tests/SonarAnalyzer.Test/TestCases/SpecifyRouteAttribute.CSharp12.cs

```diff
@@ -190,3 +190,27 @@ public class DerivedController : Controller { }
 
 [Controller]
 public class Endpoint { }
+
+[Route("api/[controller]")]
+public class ControllerWithRouteAttribute : Controller { }
+
+public class ControllerWithInheritedRoute : ControllerWithRouteAttribute            // Compliant, attribute is inherited
+{
+    [HttpGet("Test")]
+    public string Index() => "Hi!";
+}
+
+public class BaseControllerWithActionWithRoute : Controller                         // Noncompliant
+{
+    [HttpGet("Test")]
+    public virtual string Index() => "Hi!";                                         // Secondary
+
+    public virtual string Index(int id) => "Hi!";                                   // Compliant
+}
+
+public class ControllerOverridesActionWithRoute : BaseControllerWithActionWithRoute // Noncompliant
+{
+    public override string Index() => "Hi!";                                        // Secondary
```

Comment:
```suggestion
    //Route: /Test (AmbiguousMatchException raised because the base method is also in scope)
    public override string Index() => "Hi!";                                        // Secondary
```